src/Databases/DuckDB/fetchDuckDBTableStructure.cpp

```diff
@@ -0,0 +1,113 @@
+#include <Databases/DuckDB/fetchDuckDBTableStructure.h>
+
+#if USE_DUCKDB
+
+#include <Common/quoteString.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeFactory.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesDecimal.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Poco/String.h>
+
+#include <string_view>
+
+
+using namespace std::literals;
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int DUCKDB_ENGINE_ERROR;
+}
+
+static DataTypePtr convertDuckDBDataType(String type)
+{
+    DataTypePtr res;
+    type = Poco::toLower(type);
+
+    if (type == "tinyint")
+        res = std::make_shared<DataTypeInt8>();
+    else if (type == "smallint")
+        res = std::make_shared<DataTypeInt16>();
+    else if ((type.starts_with("int") && type != "int8") || type == "mediumint")
+        res = std::make_shared<DataTypeInt32>();
+    else if (type == "bigint" || type == "int8")
+        res = std::make_shared<DataTypeInt64>();
+    else if (type == "float")
```

Comment:
Probably we can support `HUGEINT` as `Int128`. Also as I see in https://duckdb.org/docs/sql/data_types/numeric.html there're unsigned `UTINYINT` / `USMALLINT` / `UINTEGER` and `UBIGINT`